Holovibes/sources/api/transform_api.cc

```diff
@@ -0,0 +1,270 @@
+}
+
+void update_time_stride(const uint time_stride)
+{
+    if (get_compute_mode() == Computation::Raw || get_import_type() == ImportType::None)
+        return;
+
+    if (time_stride == get_time_stride())
+        return;
+
+    set_time_stride(time_stride);
+    get_compute_pipe()->request(ICS::UpdateTimeStride);
+}
+
+#pragma endregion
+
+#pragma region Space Tr.
+
+void set_lambda(float value)
+{
+    if (api::get_compute_mode() == Computation::Raw)
+        return;
+
+    UPDATE_SETTING(Lambda, value < 0 ? 0 : value);
+    pipe_refresh();
+}
+
+void set_z_distance(float value)
+{
+    if (get_compute_mode() == Computation::Raw)
+        return;
+
+    // Avoid 0 for cuda kernel
+    if (value <= 0)
+        value = 0.000001f;
+
+    UPDATE_SETTING(ZDistance, value);
+    pipe_refresh();
+}
+
+void set_space_transformation(const SpaceTransformation value)
+{
+    if (api::get_compute_mode() == Computation::Raw || api::get_space_transformation() == value)
+        return;
+
+    UPDATE_SETTING(SpaceTransformation, value);
+    pipe_refresh();
+}
+
+#pragma endregion
+
+#pragma region Time Tr.
+
+void set_time_transformation(const TimeTransformation value)
+{
+    if (api::get_compute_mode() == Computation::Raw || api::get_time_transformation() == value)
+        return;
+
+    UPDATE_SETTING(TimeTransformation, value);
+    set_z_fft_shift(value == TimeTransformation::STFT);
+    get_compute_pipe()->request(ICS::UpdateTimeTransformationAlgorithm);
+}
+
+void update_time_transformation_size(uint time_transformation_size)
+{
+    if (get_compute_mode() == Computation::Raw || get_import_type() == ImportType::None)
+        return;
+
+    if (time_transformation_size == api::get_time_transformation_size())
+        return;
+
+    if (time_transformation_size < 1)
+        time_transformation_size = 1;
+
+    set_time_transformation_size(time_transformation_size);
+    get_compute_pipe()->request(ICS::UpdateTimeTransformationSize);
+}
+
+void set_x_accu_level(uint x_value)
+{
+    SET_SETTING(X, width, x_value);
+    pipe_refresh();
+}
+
+void set_x_cuts(uint value)
+{
+    if (value < get_fd().width)
+    {
+        SET_SETTING(X, start, value);
+        pipe_refresh();
+    }
+}
+
+void set_y_accu_level(uint y_value)
+{
+    SET_SETTING(Y, width, y_value);
+    pipe_refresh();
+}
+
+void set_y_cuts(uint value)
+{
+    if (value < get_fd().height)
+    {
+        SET_SETTING(Y, start, value);
+        pipe_refresh();
+    }
+}
+
+void set_x_y(uint x, uint y)
+{
+    if (get_compute_mode() == Computation::Raw || get_import_type() == ImportType::None)
+        return;
+
+    if (x < get_fd().width)
+        SET_SETTING(X, start, x);
+
+    if (y < get_fd().width)
+        SET_SETTING(Y, start, y);
+
+    pipe_refresh();
+}
+
+void set_q_index(uint value)
+{
+    SET_SETTING(Q, start, value);
+    pipe_refresh();
+}
+
+void set_q_accu_level(uint value)
+{
+    SET_SETTING(Q, width, value);
+    pipe_refresh();
+}
+
```

Comment:
Could you add :
 if (api::get_compute_mode() == Computation::Raw)
        return;